api/src/main/java/com/khali/api3/controllers/UserController.java

```diff
@@ -0,0 +1,67 @@
+package com.khali.api3.controller;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.khali.api3.repositories.UserRepository;
+import com.khali.api3.domain.user.User;
+
+import jakarta.persistence.EntityNotFoundException;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserController(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @GetMapping
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public User getUserById(@PathVariable Long id) {
+        return userRepository.findById(id)
+            .orElseThrow(() -> new EntityNotFoundException("User not found with id: " + id));
+    }
+
+    @PostMapping
+    public User createUser(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
```

Comment:
Esse comentário abre um bloco de transação dentro do banco para executar a operação.
caso a operação não dê erro o conteúdo será persistido normalmente. Mas no caso de um erro em qualquer parte da operação, nenhuma informação incompleta será salva.

se operação der erro o banco da um rollback e fecha a transação

```suggestion

@Transactional
```

acredito não ser necessário em métodos de save simples, mas em casos de updates talvez seja recomendável.
com essas informações, você decide se vai ou não aplicar o comentários em métodos de updates e métodos mais complexos.
> recomendo usa-las

caso fique interessado, tem mais alguns parâmetros dentro dessa anotação que podem ser usados